src/utils/account/account.service.ts

```diff
@@ -44,12 +44,17 @@ export class AccountService {
     await this.validateAccountCreationRequest(userData);
 
     const accountUpn = this.buildUPN(userData);
-    const result = await this.msGraphService.createUser(userData, accountUpn);
-    // Save the accountUpn on the user profile
-    if (!result)
-      throw new Error(
-        `Unable to complete account creation for ${userData.email}`
-      );
+    try {
+      const result = await this.msGraphService.createUser(userData, accountUpn);
+      if (!result)
+        throw new Error(
+          `Unable to complete account creation for ${userData.email} using UPN: ${accountUpn}`
+        );
+    } catch (e) {
+      const msg = `Unable to complete account creation for ${userData.email}: ${e}`;
+      this.logger.error(msg);
```

Comment:
If you throw an Error it will be logged, no need to log it separately.

Reply:
ok, will fix

Reply:
Wanted to add an additional error msg that account creation failed, calling an external service so want to catch and log here and then throw further. Updated to rethrow and not throw a new msg.